subprojects/build-init/src/main/groovy/org/gradle/buildinit/plugins/internal/maven/Maven2Gradle.groovy

```diff
@@ -343,16 +423,37 @@ version = '$project.version'""";
         return build.toString();
     }
 
-    def compilerSettings = { project, indent ->
+    def compilerSettings(project, indent, useKotlinAccessors = true) {
         def configuration = plugin('maven-compiler-plugin', project).configuration
-        def settings = new StringBuilder()
-        settings.append "sourceCompatibility = ${configuration.source.text() ?: '1.5'}\n"
-        settings.append "${indent}targetCompatibility = ${configuration.target.text() ?: '1.5'}\n"
         def encoding = project.properties.'project.build.sourceEncoding'.text()
-        if (encoding) {
-            settings.append "${indent}tasks.withType(JavaCompile) {\n"
-            settings.append "${indent}\toptions.encoding = '${encoding}'\n"
-            settings.append "${indent}}\n"
+        def settings = new StringBuilder()
+        switch (scriptDsl) {
+            case KOTLIN:
+                if (useKotlinAccessors) {
+                    settings.append "${indent}java {\n"
+                } else {
+                    settings.append "${indent}configure<JavaPluginConvention> {\n"
+                }
+                JavaVersion sourceCompatibility = configuration.source.text() ? JavaVersion.toVersion(configuration.source.text()) : JavaVersion.VERSION_1_5
+                JavaVersion targetCompatibility = configuration.target.text() ? JavaVersion.toVersion(configuration.target.text()) : JavaVersion.VERSION_1_5
+                settings.append "${indent}    sourceCompatibility = JavaVersion.${sourceCompatibility.name()}\n"
+                settings.append "${indent}    targetCompatibility = JavaVersion.${targetCompatibility.name()}\n"
+                settings.append "${indent}}\n"
+                if (encoding) {
+                    settings.append "${indent}tasks.withType(JavaCompile::class.java) {\n"
```

Comment:
This line can be replaced with:
```kotlin
tasks.withType<JavaCompile> {
```

Reply:
Good call! Fixed

Reply:
You are probably looking at the code in the outdated context of your original comment.

Reply:
You're right.